modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/store/SdkObjectStoreAdapter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.store;
+
+import static java.lang.String.format;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+
+import org.mule.runtime.api.store.ObjectStore;
+import org.mule.runtime.api.store.ObjectStoreException;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Adapts the Mule api for object store {@link ObjectStore} into the SDK api {@link org.mule.sdk.api.store.ObjectStore}
+ *
+ * @since 4.5.0
+ */
+public class SdkObjectStoreAdapter<T extends Serializable> implements org.mule.sdk.api.store.ObjectStore<T> {
```

Comment:
this class transforms a real objectstore into an SDK one, which means that thrown exceptions are going to be SDK ones, while the rest of the Mule code is going to be expecting Mule exceptions. Where is that being handled?

Reply:
Yes, in fact, the exceptions thrown by the Mule Runtime Object Store will be converted into the SDK OS exceptions. 
I see your point and I remember discussing the topic of exceptions within the team. We discussed making the SDK Object Store exceptions to extend the Mule API Object Store exceptions, but then we discarded the idea because that created a dependency between the SDK and Mule APIs, which is something that should be avoided as part of the forward compatibility feature, right? Do you see any other way to handle this?